graylog2-server/src/main/java/org/graylog2/indexer/cluster/Cluster.java

```diff
@@ -17,69 +17,82 @@
-import org.elasticsearch.client.Client;
-import org.elasticsearch.client.ClusterAdminClient;
-import org.elasticsearch.cluster.health.ClusterHealthStatus;
-import org.elasticsearch.cluster.node.DiscoveryNode;
+import com.google.common.primitives.Ints;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.searchbox.client.JestClient;
+import io.searchbox.client.JestResult;
+import io.searchbox.cluster.Health;
+import io.searchbox.cluster.NodesInfo;
+import io.searchbox.core.Cat;
+import io.searchbox.core.CatResult;
 import org.graylog2.indexer.IndexSetRegistry;
 import org.graylog2.indexer.esplugin.ClusterStateMonitor;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import javax.annotation.Nullable;
 import javax.inject.Inject;
 import javax.inject.Named;
 import javax.inject.Singleton;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
 import java.util.Map;
+import java.util.Optional;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 import java.util.concurrent.atomic.AtomicReference;
-import java.util.stream.Collectors;
 
 
 @Singleton
 public class Cluster {
     private static final Logger LOG = LoggerFactory.getLogger(Cluster.class);
 
-    private final Client c;
+    private final JestClient jestClient;
     private final IndexSetRegistry indexSetRegistry;
     private final ScheduledExecutorService scheduler;
     private final Duration requestTimeout;
-    private final AtomicReference<Map<String, DiscoveryNode>> nodes = new AtomicReference<>();
+    private final AtomicReference<Map> nodes = new AtomicReference<>();
 
     @Inject
-    public Cluster(Client client,
+    public Cluster(JestClient jestClient,
                    IndexSetRegistry indexSetRegistry,
                    @Named("daemonScheduler") ScheduledExecutorService scheduler,
                    @Named("elasticsearch_request_timeout") Duration requestTimeout) {
         this.scheduler = scheduler;
-        this.c = client;
+        this.jestClient = jestClient;
         this.indexSetRegistry = indexSetRegistry;
         this.requestTimeout = requestTimeout;
         // unfortunately we can't use guice here, because elasticsearch and graylog2 use different injectors and we can't
         // get to the instance to bridge.
         ClusterStateMonitor.setCluster(this);
     }
 
+    @Nullable
+    private JsonObject clusterHealth(Collection<? extends String> indices) {
+        final Health request = new Health.Builder()
+                .addIndex(indices)
+                .build();
+        try {
+            return jestClient.execute(request).getJsonObject();
+        } catch (IOException e) {
+            LOG.error("Couldn't read cluster health", e);
+            return null;
```

Comment:
Maybe we should throw a subclass of the `ElasticsearchException` wrapping the caught exception instead of returning `null`? This way, possibly important information about the fault gets lost.